runtimebp/cpu.go

```diff
@@ -19,81 +19,34 @@ import (
 // it will fallback to runtime.NumCPU() instead.
 //
 // When fallback happens, it also prints the reason to stderr.
```

Comment:
you changed the behavior so the comments should also be updated. you also removed the logging when fallback happens, which I think can still be useful.

also this function, as a public API from `runtimebp`, is supposed to provide container-aware version of `runtime.NumCPU`. if a container requested 2 cpu running on a 48 cpu physical machine, is 48 the expected/correct return for this api? maybe we just want to change the behavior of `runtimebp.GOMAXPROCS`, but keep `runtimebp.NumCPU` intact (we can have some refactoring to reduce code duplication, if we do believe GOMAXPROCS should not fallback to shares).

Reply:
Yes, "request" is a Kubernetes-specific concept. In Kubernetes, Request affects scheduling, and CFS shares as a side effect. It's intended that containers are allowed to use any CPU more than their "request" in the Kubernetes context. If there are competing containers, CFS will proportionally share the "extra" CPU time between containers based on those requests.

Since we don't know what the intended use the application is, we should not second guess and set limits that would be a major change to typical Go runtime.

Keeping the reading of quota, which is "limit" in a Kubernetes context is what we want to keep. With hard quota / limit, we have a much better idea on how many CPUs are intended to be given to the app, therefore overriding the usual number of runtime threads is a good idea.

Reply:
Updated the docs.